src/pybind/mgr/nfs/export.py

```diff
@@ -543,7 +543,26 @@ def _change_export(self, cluster_id: str, export: Dict) -> Dict[str, str]:
         except Exception as ex:
             msg = f'Failed to apply export: {ex}'
             log.exception(msg)
-            return {"state": "error", "msg": msg}
+
+            old_export = self._fetch_export(cluster_id, export['pseudo'])
+            if old_export:
+                # without this, if a try is made to modify export_id then,
+                # although it won't be allowed but the err dict will show
+                # the export_id from the supplied export block.
+                export_id = old_export.export_id
+            else:
+                # it's a request to create a new export, two cases:
+                # 1) export block contains export_id: fetching directly it
+                # from the supplied export block should work.
+                # 2) export block doesn't contain export_id: get the next
+                # available val, this is safe since we are here because
+                # of an error occurred in create_export_from_dict() and the
+                # problematic export won't have been saved in exports dict
+                # i.e. _save_export() won't have been called.
+                export_id = export['export_id'] if export.get('export_id') \
+                    else self._gen_export_id(cluster_id)
+
+            return {"export_id": export_id, "state": "error", "msg": msg}
```

Comment:
Why is the `state` field useful?

Reply:
the -EIO errno is telling that (some) exports failed updation - isn't that enough?

Reply:
> the -EIO errno is telling that (some) exports failed updation - isn't that enough?

found it, it has a utility in `class AppliedExportResults` where the `status` is used to return EIO

Reply:
> Why is the `state` field useful?

I've get an idea, we can get rid of this `state` by changing `if value.get("state", "") == "error"` to `if value.pop("state", "") == "error"`in 

https://github.com/ceph/ceph/blob/9d7c18257836dc888b4a300f3b5af9f080910986/src/pybind/mgr/nfs/export.py#L171-L173

this way it would just be:
```
[
  {
    "export_id": 1,
    "msg": "Failed to apply export: filesystem foo not found",
  },
  {
    "export_id": 2,
    "msg": "Failed to apply export: export FSAL user_id must be 'nfs.my-nfs.2'",
  }
]
```
and this looks really neat

Reply:
@vshankar @phlogistonjohn what do you think?

Reply:
I think there should be a clear indicator for each item showing if it was ok or not, for automation code that might be consuming the JSON. Code needs to easily distinguish between errors and non-errors. That could be via state field or via always having an 'errno' field on error items. But requiring code to scrape text is not a good approach and defeats the point of using JSON output.

Reply:
@phlogistonjohn Agree. So we agree to keep the `state` field and possible values for those are `success` or `error`.

Reply:
@vshankar possible values for `state` can be `added`/`updated`/`error`

(FYI `ceph nfs export apply -i <{json/conf}_file>`) not only can update an export but create a new export too.

Reply:
Fair enough. Although `updated` it pretty much depicting the same as `added`, I'm ok to separate those out. 👍

Reply:
Do we have tests that check the `state` field?

Reply:
It makes sense to add the changes here (state+test) IMO.

Reply:
the tracker is of adding export_id to err dict, i dont think the test case is relevant to the tracker. the test case will be relevant in the tracker where i'll write the code to fix the errno and the err string, where multiple blocks will be supplied and the response could be added/error.

Reply:
will be added in https://github.com/ceph/ceph/pull/53431